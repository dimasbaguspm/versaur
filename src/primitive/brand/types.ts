import type { HTMLAttributes } from 'react'

/**
 * Props for the Brand component
 */
export interface BrandProps extends HTMLAttributes<HTMLSpanElement> {
  /**
   * The app name to display (predefined, default: 'spenicle')
   */
  name: 'spenicle' | 'hub'
  /**
   * The size of the brand icon and text
   * @default 'md'
   */
  size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl'
  shape?: 'square' | 'rounded' | 'circle'
}

export type BrandAtomProps = Pick<BrandProps, 'shape' | 'size' | 'name'>
