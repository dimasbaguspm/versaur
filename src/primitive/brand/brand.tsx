/// <reference types="vite-plugin-svgr/client" />

import type { BrandProps } from './types'
import { cn } from '@/utils/cn'
import { BrandLogo } from './brand.atoms'
import { forwardRef } from 'react'

/**
 * Brand component for displaying app identity icon and name
 *
 * @param {BrandProps} props - Brand component props
 * @returns {JSX.Element}
 */
export const Brand = forwardRef<HTMLDivElement, BrandProps>(
  ({ name: _name = 'spenicle', size = 'md', shape, ...props }, ref) => {
    return (
      <span
        ref={ref}
        className={cn('inline-flex items-center gap-2')}
        {...props}
      >
        <BrandLogo name={_name} shape={shape} size={size} aria-hidden='true' />
      </span>
    )
  }
)
