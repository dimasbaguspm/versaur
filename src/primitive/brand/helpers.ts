import { cva } from 'class-variance-authority'

/**
 * CVA for brand icon size
 * Extends IconProps['size']
 */
export const brandSizeClass = cva('', {
  variants: {
    size: {
      xs: 'h-6 w-6',
      sm: 'h-7 w-7',
      md: 'h-9 w-9',
      lg: 'h-10 w-10',
      xl: 'h-12 w-12',
    },
  },
  defaultVariants: {
    size: 'md',
  },
})
