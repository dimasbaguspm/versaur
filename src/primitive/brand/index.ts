export { Brand } from './brand'
export type * from './types'
