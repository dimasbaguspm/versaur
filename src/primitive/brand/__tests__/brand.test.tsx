import { render } from '@testing-library/react'
import { composeStories } from '@storybook/react'
import * as stories from '../brand.stories'

describe('Brand', () => {
  const { Default, Name, Sizes } = composeStories(stories)

  it('renders default brand (rounded)', () => {
    const { container } = render(<Default />)
    expect(container.firstChild).toBeInTheDocument()
    expect(container.querySelector('svg')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('renders all shape variants in Name story', () => {
    const { container } = render(<Name />)
    const svgs = container.querySelectorAll('svg')
    expect(svgs.length).toBe(6)
    expect(container).toMatchSnapshot()
  })

  it('renders all size variants in Sizes story', () => {
    const { container } = render(<Sizes />)
    const svgs = container.querySelectorAll('svg')
    expect(svgs.length).toBe(10)
    expect(container).toMatchSnapshot()
  })
})
