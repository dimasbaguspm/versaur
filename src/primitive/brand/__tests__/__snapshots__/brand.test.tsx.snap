// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Brand > renders all shape variants in Name story 1`] = `
<div>
  <div
    class="flex gap-6 items-center"
  >
    <span
      class="inline-flex items-center gap-2"
    >
      <svg
        aria-hidden="true"
        class="inline-block h-9 w-9"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Spenicle
        </title>
        <rect
          fill="#e07a5f"
          height="32"
          width="32"
        />
        <g
          transform="translate(16, 16) scale(0.625) translate(-12, -12)"
        >
          <path
            d="M4 14a1 1 0 0 1-.78-1.63l9.9-10.2a.5.5 0 0 1 .86.46l-1.92 6.02A1 1 0 0 0 13 10h7a1 1 0 0 1 .78 1.63l-9.9 10.2a.5.5 0 0 1-.86-.46l1.92-6.02A1 1 0 0 0 11 14z"
            fill="white"
          />
        </g>
      </svg>
    </span>
    <span
      class="inline-flex items-center gap-2"
    >
      <svg
        aria-hidden="true"
        class="inline-block h-9 w-9"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Spenicle
        </title>
        <rect
          fill="#e07a5f"
          height="32"
          rx="8"
          width="32"
        />
        <g
          transform="translate(16, 16) scale(0.625) translate(-12, -12)"
        >
          <path
            d="M4 14a1 1 0 0 1-.78-1.63l9.9-10.2a.5.5 0 0 1 .86.46l-1.92 6.02A1 1 0 0 0 13 10h7a1 1 0 0 1 .78 1.63l-9.9 10.2a.5.5 0 0 1-.86-.46l1.92-6.02A1 1 0 0 0 11 14z"
            fill="white"
          />
        </g>
      </svg>
    </span>
    <span
      class="inline-flex items-center gap-2"
    >
      <svg
        aria-hidden="true"
        class="inline-block h-9 w-9"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Spenicle
        </title>
        <circle
          cx="16"
          cy="16"
          fill="#e07a5f"
          r="16"
        />
        <g
          transform="translate(16, 16) scale(0.625) translate(-12, -12)"
        >
          <path
            d="M4 14a1 1 0 0 1-.78-1.63l9.9-10.2a.5.5 0 0 1 .86.46l-1.92 6.02A1 1 0 0 0 13 10h7a1 1 0 0 1 .78 1.63l-9.9 10.2a.5.5 0 0 1-.86-.46l1.92-6.02A1 1 0 0 0 11 14z"
            fill="white"
          />
        </g>
      </svg>
    </span>
    <span
      class="inline-flex items-center gap-2"
    >
      <svg
        aria-hidden="true"
        class="inline-block h-9 w-9"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Hub (Square)
        </title>
        <rect
          fill="#e07a5f"
          height="32"
          rx="0"
          width="32"
          x="0"
          y="0"
        />
        <g
          transform="translate(16, 16) scale(0.625) translate(-12, -12)"
        >
          <path
            d="M20.341 6.484A10 10 0 0 1 10.266 21.85"
            fill="none"
            stroke="white"
            stroke-width="2"
          />
          <path
            d="M3.659 17.516A10 10 0 0 1 13.74 2.152"
            fill="none"
            stroke="white"
            stroke-width="2"
          />
          <circle
            cx="12"
            cy="12"
            fill="white"
            r="3"
          />
          <circle
            cx="19"
            cy="5"
            fill="white"
            r="2"
          />
          <circle
            cx="5"
            cy="19"
            fill="white"
            r="2"
          />
        </g>
      </svg>
    </span>
    <span
      class="inline-flex items-center gap-2"
    >
      <svg
        aria-hidden="true"
        class="inline-block h-9 w-9"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Hub (Rounded)
        </title>
        <rect
          fill="#e07a5f"
          height="28"
          rx="8"
          width="28"
          x="2"
          y="2"
        />
        <g
          transform="translate(16, 16) scale(0.625) translate(-12, -12)"
        >
          <path
            d="M20.341 6.484A10 10 0 0 1 10.266 21.85"
            fill="none"
            stroke="white"
            stroke-width="2"
          />
          <path
            d="M3.659 17.516A10 10 0 0 1 13.74 2.152"
            fill="none"
            stroke="white"
            stroke-width="2"
          />
          <circle
            cx="12"
            cy="12"
            fill="white"
            r="3"
          />
          <circle
            cx="19"
            cy="5"
            fill="white"
            r="2"
          />
          <circle
            cx="5"
            cy="19"
            fill="white"
            r="2"
          />
        </g>
      </svg>
    </span>
    <span
      class="inline-flex items-center gap-2"
    >
      <svg
        aria-hidden="true"
        class="inline-block h-9 w-9"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Hub (Circle)
        </title>
        <circle
          cx="16"
          cy="16"
          fill="#e07a5f"
          r="16"
        />
        <g
          transform="translate(16, 16) scale(0.625) translate(-12, -12)"
        >
          <path
            d="M20.341 6.484A10 10 0 0 1 10.266 21.85"
            fill="none"
            stroke="white"
            stroke-width="2"
          />
          <path
            d="M3.659 17.516A10 10 0 0 1 13.74 2.152"
            fill="none"
            stroke="white"
            stroke-width="2"
          />
          <circle
            cx="12"
            cy="12"
            fill="white"
            r="3"
          />
          <circle
            cx="19"
            cy="5"
            fill="white"
            r="2"
          />
          <circle
            cx="5"
            cy="19"
            fill="white"
            r="2"
          />
        </g>
      </svg>
    </span>
    <span
      class="inline-flex items-center gap-2"
    >
      <svg
        aria-hidden="true"
        class="inline-block h-9 w-9"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Notunic
        </title>
        <rect
          fill="#e07a5f"
          height="32"
          width="32"
        />
        <g
          transform="translate(16, 16) scale(0.625) translate(-12, -12)"
        >
          <path
            d="M13.4 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2v-7.4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 6h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 10h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 14h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 18h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M21.378 5.626a1 1 0 1 0-3.004-3.004l-5.01 5.012a2 2 0 0 0-.506.854l-.837 2.87a.5.5 0 0 0 .62.62l2.87-.837a2 2 0 0 0 .854-.506z"
            fill="white"
          />
        </g>
      </svg>
    </span>
    <span
      class="inline-flex items-center gap-2"
    >
      <svg
        aria-hidden="true"
        class="inline-block h-9 w-9"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Notunic
        </title>
        <rect
          fill="#e07a5f"
          height="32"
          rx="8"
          width="32"
        />
        <g
          transform="translate(16, 16) scale(0.625) translate(-12, -12)"
        >
          <path
            d="M13.4 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2v-7.4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 6h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 10h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 14h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 18h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M21.378 5.626a1 1 0 1 0-3.004-3.004l-5.01 5.012a2 2 0 0 0-.506.854l-.837 2.87a.5.5 0 0 0 .62.62l2.87-.837a2 2 0 0 0 .854-.506z"
            fill="white"
          />
        </g>
      </svg>
    </span>
    <span
      class="inline-flex items-center gap-2"
    >
      <svg
        aria-hidden="true"
        class="inline-block h-9 w-9"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Notunic
        </title>
        <circle
          cx="16"
          cy="16"
          fill="#e07a5f"
          r="16"
        />
        <g
          transform="translate(16, 16) scale(0.625) translate(-12, -12)"
        >
          <path
            d="M13.4 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2v-7.4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 6h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 10h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 14h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 18h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M21.378 5.626a1 1 0 1 0-3.004-3.004l-5.01 5.012a2 2 0 0 0-.506.854l-.837 2.87a.5.5 0 0 0 .62.62l2.87-.837a2 2 0 0 0 .854-.506z"
            fill="white"
          />
        </g>
      </svg>
    </span>
  </div>
</div>
`;

exports[`Brand > renders all size variants in Sizes story 1`] = `
<div>
  <div
    class="flex flex-col gap-4"
  >
    <span
      class="inline-flex items-center gap-2"
    >
      <svg
        aria-hidden="true"
        class="inline-block h-6 w-6"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Spenicle
        </title>
        <rect
          fill="#e07a5f"
          height="32"
          width="32"
        />
        <g
          transform="translate(16, 16) scale(0.625) translate(-12, -12)"
        >
          <path
            d="M4 14a1 1 0 0 1-.78-1.63l9.9-10.2a.5.5 0 0 1 .86.46l-1.92 6.02A1 1 0 0 0 13 10h7a1 1 0 0 1 .78 1.63l-9.9 10.2a.5.5 0 0 1-.86-.46l1.92-6.02A1 1 0 0 0 11 14z"
            fill="white"
          />
        </g>
      </svg>
    </span>
    <span
      class="inline-flex items-center gap-2"
    >
      <svg
        aria-hidden="true"
        class="inline-block h-7 w-7"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Spenicle
        </title>
        <rect
          fill="#e07a5f"
          height="32"
          width="32"
        />
        <g
          transform="translate(16, 16) scale(0.625) translate(-12, -12)"
        >
          <path
            d="M4 14a1 1 0 0 1-.78-1.63l9.9-10.2a.5.5 0 0 1 .86.46l-1.92 6.02A1 1 0 0 0 13 10h7a1 1 0 0 1 .78 1.63l-9.9 10.2a.5.5 0 0 1-.86-.46l1.92-6.02A1 1 0 0 0 11 14z"
            fill="white"
          />
        </g>
      </svg>
    </span>
    <span
      class="inline-flex items-center gap-2"
    >
      <svg
        aria-hidden="true"
        class="inline-block h-9 w-9"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Spenicle
        </title>
        <rect
          fill="#e07a5f"
          height="32"
          width="32"
        />
        <g
          transform="translate(16, 16) scale(0.625) translate(-12, -12)"
        >
          <path
            d="M4 14a1 1 0 0 1-.78-1.63l9.9-10.2a.5.5 0 0 1 .86.46l-1.92 6.02A1 1 0 0 0 13 10h7a1 1 0 0 1 .78 1.63l-9.9 10.2a.5.5 0 0 1-.86-.46l1.92-6.02A1 1 0 0 0 11 14z"
            fill="white"
          />
        </g>
      </svg>
    </span>
    <span
      class="inline-flex items-center gap-2"
    >
      <svg
        aria-hidden="true"
        class="inline-block h-10 w-10"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Spenicle
        </title>
        <rect
          fill="#e07a5f"
          height="32"
          width="32"
        />
        <g
          transform="translate(16, 16) scale(0.625) translate(-12, -12)"
        >
          <path
            d="M4 14a1 1 0 0 1-.78-1.63l9.9-10.2a.5.5 0 0 1 .86.46l-1.92 6.02A1 1 0 0 0 13 10h7a1 1 0 0 1 .78 1.63l-9.9 10.2a.5.5 0 0 1-.86-.46l1.92-6.02A1 1 0 0 0 11 14z"
            fill="white"
          />
        </g>
      </svg>
    </span>
    <span
      class="inline-flex items-center gap-2"
    >
      <svg
        aria-hidden="true"
        class="inline-block h-12 w-12"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Spenicle
        </title>
        <rect
          fill="#e07a5f"
          height="32"
          width="32"
        />
        <g
          transform="translate(16, 16) scale(0.625) translate(-12, -12)"
        >
          <path
            d="M4 14a1 1 0 0 1-.78-1.63l9.9-10.2a.5.5 0 0 1 .86.46l-1.92 6.02A1 1 0 0 0 13 10h7a1 1 0 0 1 .78 1.63l-9.9 10.2a.5.5 0 0 1-.86-.46l1.92-6.02A1 1 0 0 0 11 14z"
            fill="white"
          />
        </g>
      </svg>
    </span>
    <span
      class="inline-flex items-center gap-2"
    >
      <svg
        aria-hidden="true"
        class="inline-block h-6 w-6"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Hub (Square)
        </title>
        <rect
          fill="#e07a5f"
          height="32"
          rx="0"
          width="32"
          x="0"
          y="0"
        />
        <g
          transform="translate(16, 16) scale(0.625) translate(-12, -12)"
        >
          <path
            d="M20.341 6.484A10 10 0 0 1 10.266 21.85"
            fill="none"
            stroke="white"
            stroke-width="2"
          />
          <path
            d="M3.659 17.516A10 10 0 0 1 13.74 2.152"
            fill="none"
            stroke="white"
            stroke-width="2"
          />
          <circle
            cx="12"
            cy="12"
            fill="white"
            r="3"
          />
          <circle
            cx="19"
            cy="5"
            fill="white"
            r="2"
          />
          <circle
            cx="5"
            cy="19"
            fill="white"
            r="2"
          />
        </g>
      </svg>
    </span>
    <span
      class="inline-flex items-center gap-2"
    >
      <svg
        aria-hidden="true"
        class="inline-block h-7 w-7"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Hub (Square)
        </title>
        <rect
          fill="#e07a5f"
          height="32"
          rx="0"
          width="32"
          x="0"
          y="0"
        />
        <g
          transform="translate(16, 16) scale(0.625) translate(-12, -12)"
        >
          <path
            d="M20.341 6.484A10 10 0 0 1 10.266 21.85"
            fill="none"
            stroke="white"
            stroke-width="2"
          />
          <path
            d="M3.659 17.516A10 10 0 0 1 13.74 2.152"
            fill="none"
            stroke="white"
            stroke-width="2"
          />
          <circle
            cx="12"
            cy="12"
            fill="white"
            r="3"
          />
          <circle
            cx="19"
            cy="5"
            fill="white"
            r="2"
          />
          <circle
            cx="5"
            cy="19"
            fill="white"
            r="2"
          />
        </g>
      </svg>
    </span>
    <span
      class="inline-flex items-center gap-2"
    >
      <svg
        aria-hidden="true"
        class="inline-block h-9 w-9"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Hub (Square)
        </title>
        <rect
          fill="#e07a5f"
          height="32"
          rx="0"
          width="32"
          x="0"
          y="0"
        />
        <g
          transform="translate(16, 16) scale(0.625) translate(-12, -12)"
        >
          <path
            d="M20.341 6.484A10 10 0 0 1 10.266 21.85"
            fill="none"
            stroke="white"
            stroke-width="2"
          />
          <path
            d="M3.659 17.516A10 10 0 0 1 13.74 2.152"
            fill="none"
            stroke="white"
            stroke-width="2"
          />
          <circle
            cx="12"
            cy="12"
            fill="white"
            r="3"
          />
          <circle
            cx="19"
            cy="5"
            fill="white"
            r="2"
          />
          <circle
            cx="5"
            cy="19"
            fill="white"
            r="2"
          />
        </g>
      </svg>
    </span>
    <span
      class="inline-flex items-center gap-2"
    >
      <svg
        aria-hidden="true"
        class="inline-block h-10 w-10"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Hub (Square)
        </title>
        <rect
          fill="#e07a5f"
          height="32"
          rx="0"
          width="32"
          x="0"
          y="0"
        />
        <g
          transform="translate(16, 16) scale(0.625) translate(-12, -12)"
        >
          <path
            d="M20.341 6.484A10 10 0 0 1 10.266 21.85"
            fill="none"
            stroke="white"
            stroke-width="2"
          />
          <path
            d="M3.659 17.516A10 10 0 0 1 13.74 2.152"
            fill="none"
            stroke="white"
            stroke-width="2"
          />
          <circle
            cx="12"
            cy="12"
            fill="white"
            r="3"
          />
          <circle
            cx="19"
            cy="5"
            fill="white"
            r="2"
          />
          <circle
            cx="5"
            cy="19"
            fill="white"
            r="2"
          />
        </g>
      </svg>
    </span>
    <span
      class="inline-flex items-center gap-2"
    >
      <svg
        aria-hidden="true"
        class="inline-block h-12 w-12"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Hub (Square)
        </title>
        <rect
          fill="#e07a5f"
          height="32"
          rx="0"
          width="32"
          x="0"
          y="0"
        />
        <g
          transform="translate(16, 16) scale(0.625) translate(-12, -12)"
        >
          <path
            d="M20.341 6.484A10 10 0 0 1 10.266 21.85"
            fill="none"
            stroke="white"
            stroke-width="2"
          />
          <path
            d="M3.659 17.516A10 10 0 0 1 13.74 2.152"
            fill="none"
            stroke="white"
            stroke-width="2"
          />
          <circle
            cx="12"
            cy="12"
            fill="white"
            r="3"
          />
          <circle
            cx="19"
            cy="5"
            fill="white"
            r="2"
          />
          <circle
            cx="5"
            cy="19"
            fill="white"
            r="2"
          />
        </g>
      </svg>
    </span>
    <span
      class="inline-flex items-center gap-2"
    >
      <svg
        aria-hidden="true"
        class="inline-block h-6 w-6"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Notunic
        </title>
        <rect
          fill="#e07a5f"
          height="32"
          width="32"
        />
        <g
          transform="translate(16, 16) scale(0.625) translate(-12, -12)"
        >
          <path
            d="M13.4 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2v-7.4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 6h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 10h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 14h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 18h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M21.378 5.626a1 1 0 1 0-3.004-3.004l-5.01 5.012a2 2 0 0 0-.506.854l-.837 2.87a.5.5 0 0 0 .62.62l2.87-.837a2 2 0 0 0 .854-.506z"
            fill="white"
          />
        </g>
      </svg>
    </span>
    <span
      class="inline-flex items-center gap-2"
    >
      <svg
        aria-hidden="true"
        class="inline-block h-7 w-7"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Notunic
        </title>
        <rect
          fill="#e07a5f"
          height="32"
          width="32"
        />
        <g
          transform="translate(16, 16) scale(0.625) translate(-12, -12)"
        >
          <path
            d="M13.4 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2v-7.4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 6h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 10h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 14h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 18h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M21.378 5.626a1 1 0 1 0-3.004-3.004l-5.01 5.012a2 2 0 0 0-.506.854l-.837 2.87a.5.5 0 0 0 .62.62l2.87-.837a2 2 0 0 0 .854-.506z"
            fill="white"
          />
        </g>
      </svg>
    </span>
    <span
      class="inline-flex items-center gap-2"
    >
      <svg
        aria-hidden="true"
        class="inline-block h-9 w-9"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Notunic
        </title>
        <rect
          fill="#e07a5f"
          height="32"
          width="32"
        />
        <g
          transform="translate(16, 16) scale(0.625) translate(-12, -12)"
        >
          <path
            d="M13.4 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2v-7.4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 6h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 10h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 14h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 18h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M21.378 5.626a1 1 0 1 0-3.004-3.004l-5.01 5.012a2 2 0 0 0-.506.854l-.837 2.87a.5.5 0 0 0 .62.62l2.87-.837a2 2 0 0 0 .854-.506z"
            fill="white"
          />
        </g>
      </svg>
    </span>
    <span
      class="inline-flex items-center gap-2"
    >
      <svg
        aria-hidden="true"
        class="inline-block h-10 w-10"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Notunic
        </title>
        <rect
          fill="#e07a5f"
          height="32"
          width="32"
        />
        <g
          transform="translate(16, 16) scale(0.625) translate(-12, -12)"
        >
          <path
            d="M13.4 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2v-7.4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 6h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 10h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 14h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 18h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M21.378 5.626a1 1 0 1 0-3.004-3.004l-5.01 5.012a2 2 0 0 0-.506.854l-.837 2.87a.5.5 0 0 0 .62.62l2.87-.837a2 2 0 0 0 .854-.506z"
            fill="white"
          />
        </g>
      </svg>
    </span>
    <span
      class="inline-flex items-center gap-2"
    >
      <svg
        aria-hidden="true"
        class="inline-block h-12 w-12"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Notunic
        </title>
        <rect
          fill="#e07a5f"
          height="32"
          width="32"
        />
        <g
          transform="translate(16, 16) scale(0.625) translate(-12, -12)"
        >
          <path
            d="M13.4 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2v-7.4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 6h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 10h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 14h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M2 18h4"
            fill="none"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M21.378 5.626a1 1 0 1 0-3.004-3.004l-5.01 5.012a2 2 0 0 0-.506.854l-.837 2.87a.5.5 0 0 0 .62.62l2.87-.837a2 2 0 0 0 .854-.506z"
            fill="white"
          />
        </g>
      </svg>
    </span>
  </div>
</div>
`;

exports[`Brand > renders default brand (rounded) 1`] = `
<div>
  <span
    class="inline-flex items-center gap-2"
  >
    <svg
      aria-hidden="true"
      class="inline-block h-9 w-9"
      fill="none"
      height="32"
      viewBox="0 0 32 32"
      width="32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Spenicle
      </title>
      <rect
        fill="#e07a5f"
        height="32"
        rx="8"
        width="32"
      />
      <g
        transform="translate(16, 16) scale(0.625) translate(-12, -12)"
      >
        <path
          d="M4 14a1 1 0 0 1-.78-1.63l9.9-10.2a.5.5 0 0 1 .86.46l-1.92 6.02A1 1 0 0 0 13 10h7a1 1 0 0 1 .78 1.63l-9.9 10.2a.5.5 0 0 1-.86-.46l1.92-6.02A1 1 0 0 0 11 14z"
          fill="white"
        />
      </g>
    </svg>
  </span>
</div>
`;
